controllers/reconcile_configmap.go

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/ghodss/yaml"
+	"github.com/pkg/errors"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/v2"
+	kubeletconfigv1beta1 "k8s.io/kubelet/config/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	machineconfigv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	mcov1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+
+	rtemanifests "github.com/k8stopologyawareschedwg/deployer/pkg/manifests/rte"
+
+	nropv1alpha1 "github.com/openshift-kni/numaresources-operator/api/numaresourcesoperator/v1alpha1"
+	"github.com/openshift-kni/numaresources-operator/pkg/apply"
+	cfgstate "github.com/openshift-kni/numaresources-operator/pkg/objectstate/cfg"
+	rtestate "github.com/openshift-kni/numaresources-operator/pkg/objectstate/rte"
+	rteconfig "github.com/openshift-kni/numaresources-operator/rte/pkg/config"
+	"github.com/openshift-kni/numaresources-operator/rte/pkg/sysinfo"
+)
+
+const (
+	ConfigDataField = "config.yaml"
+)
+
+type configMapHelper struct {
+	cli       client.Client
+	scm       *runtime.Scheme
+	instance  *nropv1alpha1.NUMAResourcesOperator
+	namespace string
+}
+
+func (cmh *configMapHelper) reconcileConfigMap(ctx context.Context, kcKey client.ObjectKey) (*corev1.ConfigMap, error) {
```

Comment:
So basically this is a call from the kubelet reconcile loop right ?  not a second actual reconcile loop ?  (just trying to understand by the PR description and the code de-facto)

Reply:
We have a completely separate controller which watches kubeletconfigs and maintains configMaps which are then mounted by daemonsets, handled by the existing controller.
At this point in time, there is no sync whatsoever between the two controllers, but I think we will need some form of code sharing in the future, and anticipating this the code is arranged to allow that already